clientv3/retry_interceptor.go

```diff
@@ -105,6 +106,26 @@ func (c *Client) streamClientInterceptor(logger *zap.Logger, optFuncs ...retryOp
 	intOpts := reuseOrNewWithCallOptions(defaultOptions, optFuncs)
 	return func(ctx context.Context, desc *grpc.StreamDesc, cc *grpc.ClientConn, method string, streamer grpc.Streamer, opts ...grpc.CallOption) (grpc.ClientStream, error) {
 		ctx = withVersion(ctx)
+		// getToken automatically
+		// TODO(cfc4n): keep this code block, remove codes about getToken in client.go after pr #12165 merged.
+		var creds credentials.PerRPCCredentials
+		if c.authTokenBundle != nil {
+			// equal to c.Username != "" && c.Password != ""
+			err := c.getToken(ctx)
+			if err != nil && rpctypes.Error(err) != rpctypes.ErrAuthNotEnabled {
+				logger.Error("clientv3/retry_interceptor: getToken failed", zap.Error(err))
+				return nil, err
+			}
+			creds = c.authTokenBundle.PerRPCCredentials()
+		}
+		if creds != nil {
```

Comment:
I think logging metadata wouldn't be good, it can expose a credential in client side logs, right?

Reply:
Thanks, But I think that no other credential info in client side logs in https://github.com/etcd-io/etcd/blob/76e769ce95ca0d4d0e3486712d96956260db04b8/clientv3/credentials/credentials.go#L110-L115 ,just `authToken` only, there is not other keyword. Logging metadata  for debug .

Reply:
yeah, but if the token can be leaked and malicious person can get it, it can result security issue. I think it shouldn't be logged. How do you think?

Reply:
emmm, you are right.

 I fixed it  at https://github.com/etcd-io/etcd/commit/0649f9121ba4b91561d90dfd9d29107c07986603 .
but CI was failed at https://travis-ci.com/github/etcd-io/etcd/jobs/385106741#L2064 . 
```
=== RUN   TestBalancerUnderNetworkPartitionWatchLeader
    TestBalancerUnderNetworkPartitionWatchLeader: network_partition_test.go:266: took too long to detect leader lost
--- FAIL: TestBalancerUnderNetworkPartitionWatchLeader (3.35s)
```

But ,I test passed on my macbook like this:
```
$ PASSES=integration TESTCASE=TestBalancerUnderNetworkPartitionWatchLeader ./test                           [015eab45e]

Running with TEST_CPUS: 1,2,4
Starting 'integration' pass at 2020年 9月16日 星期三 12时26分27秒 CST
Running integration tests...
testing: warning: no tests to run
PASS
ok  	go.etcd.io/etcd/v3/integration	0.017s [no tests to run]
testing: warning: no tests to run
PASS
ok  	go.etcd.io/etcd/v3/client/integration	0.026s [no tests to run]
=== RUN   TestBalancerUnderNetworkPartitionWatchLeader
--- PASS: TestBalancerUnderNetworkPartitionWatchLeader (1.65s)
=== RUN   TestBalancerUnderNetworkPartitionWatchLeader
--- PASS: TestBalancerUnderNetworkPartitionWatchLeader (1.40s)
=== RUN   TestBalancerUnderNetworkPartitionWatchLeader
--- PASS: TestBalancerUnderNetworkPartitionWatchLeader (1.23s)
PASS
ok  	go.etcd.io/etcd/v3/clientv3/integration	4.306s
testing: warning: no tests to run
PASS
ok  	go.etcd.io/etcd/v3/contrib/raftexample	0.013s [no tests to run]
Finished 'integration' pass at 2020年 9月16日 星期三 12时26分45秒 CST
Success
```

Reply:
CI works. thank you. @mitake